File: q1.cpp
#include <iostream>

int main() {
    double price, discount, paid;
    std::cout << "Input:" << std::endl;
    std::cout << "Price: ";
    std::cin >> price;
    std::cout << "Discount: ";
    std::cin >> discount;
    std::cout << "Paid: ";
    std::cin >> paid;
    std::cout << std::endl;
    std::cout << "Output:" << std::endl;
    std::cout << "Price: " << price << std::endl;
    std::cout << "Discount: " << discount << std::endl;
    std::cout << "Discounted: " << price * (discount / 100.0) << std::endl;
    std::cout << "Net Price: " << price - price * (discount / 100.0) << std::endl;
    std::cout << "Paid: " << paid << std::endl;
    std::cout << "Change: " << paid - price + price * (discount / 100.0) << std::endl;
}

File: src/q1.rs
// basic shop
// price
// discount
// discounted
// net price
// paid
// change

use crate::input::input;

pub fn q1() {
    println!("Input:");
    let price: f64 = input(Some("Price: "));
    let discount: f64 = input(Some("Discount: "));
    let paid: f64 = input(Some("Paid: "));

    let discounted = price * (discount / 100.0);
    let net_price = price - discounted;
    let change = paid - net_price;
    println!();
    println!("Output:");
    println!("Price: {:.2}", price);
    println!("Discount: {:.2}", discount);
    println!("Discounted: {:.2}", discounted);
    println!("Net Price: {:.2}", net_price);
    println!("Paid: {:.2}", paid);
    println!("Change: {:.2}", change);
}
